x-pack/plugin/wildcard/src/main/java/org/elasticsearch/xpack/wildcard/mapper/WildcardFieldMapper.java

```diff
@@ -215,13 +302,15 @@ public WildcardFieldMapper build(BuilderContext context) {
         private WildcardFieldType(String name, FieldType fieldType, Map<String, String> meta) {
             super(name, true, fieldType.stored(), true,
                 new TextSearchInfo(fieldType, null, Lucene.KEYWORD_ANALYZER, Lucene.KEYWORD_ANALYZER), meta);
-            setIndexAnalyzer(WILDCARD_ANALYZER);
+            setIndexAnalyzer(WILDCARD_ANALYZER_7_10);
         }
 
         @Override
         public Query wildcardQuery(String wildcardPattern, RewriteMethod method, boolean caseInsensitive, QueryShardContext context) {
 
             String ngramIndexPattern = addLineEndChars(toLowerCase(wildcardPattern));
+            
+            Analyzer analyzer = getCorrectAnalyzerForIndexVersion(context);
```

Comment:
You don't need to resolve this at query-time. The correct analyzer can be picked once from the index created version when the field type is created.